### PR TITLE
[emoji-plugin] fix/mouse cursor is shown as pointer on the actual icons + fix the emoji plugin in RTL+remove the emoji plugin from mobile+refactoring the emoji modal and giphy modal to be more consistent

### PR DESCRIPTION
**Pull request by barboaron**

### CODE REVIEW

File: packages/editor-common/web/src/Utils/getModalStyles.js [hunk 1]
```diff
@@ -142,3 +142,29 @@ export const getModalStyles = ({
...
+  const { top, left, right } = buttonRef.getBoundingClientRect();
+  const isAboveButton = top - 293 > 0;
+  const isRtl = buttonRef.closest('[dir=rtl]') !== null;
+  const desktopModalStyle = {
+    top: isAboveButton ? top - 293 : top + 30,
+    margin: 0,
+    position: 'absolute',
+  };
+  if (isRtl) {
+    desktopModalStyle.right = window.innerWidth - right - 10;
```
Comment (shaulgo) [inline packages/editor-common/web/src/Utils/getModalStyles.js]: does this need window.innerWidth

File: packages/editor-common/web/src/Utils/getModalStyles.js [hunk 2]
```diff
@@ -142,3 +142,29 @@ export const getModalStyles = ({
...
+  const modalStyles = getModalStyles({
+    customStyles,
+    fullScreen,
+    inline,
+    isMobile,
+  });
+  const { top, left, right } = buttonRef.getBoundingClientRect();
+  const isAboveButton = top - 293 > 0;
+  const isRtl = buttonRef.closest('[dir=rtl]') !== null;
+  const desktopModalStyle = {
```
Comment (shaulgo) [inline packages/editor-common/web/src/Utils/getModalStyles.js]: you can change the name to contentStyles.

File: packages/plugin-emoji/web/src/constants.js [hunk 3]
```diff
@@ -62,9 +62,9 @@ export const DesktopFlyOutModalStyles = Object.freeze({
   content: {
     width: '325px',
     boxSizing: 'border-box',
-    height: '281px',
+    height: '282px',
     overflow: 'visible',
-    border: 'solid 1px #ededed',
+    border: '1px solid #ccc',
```
Comment (shaulgo) [inline packages/plugin-emoji/web/src/constants.js]: this changes are on purpose?

File: packages/plugin-emoji/web/src/toolbar/insert-buttons.jsx [hunk 4]
```diff
@@ -11,36 +11,31 @@ import EmojiPluginIcon from '../icons/EmojiPluginIcon.svg';
...
           customStyles: DesktopFlyOutModalStyles,
           fullScreen: true,
         });
-        const { top, left } = buttonRef.getBoundingClientRect();
-        const modalLeft = left - 288;
-        const isAboveButton = top - 293 > 0;
-        const modalTop = isAboveButton ? top - 293 : top + 30;
+
         return {
           ...modalStyles,
```
Comment (shaulgo) [inline packages/plugin-emoji/web/src/toolbar/insert-buttons.jsx]: this object creation can move into getBottomToolbarModalStayles?

File: packages/plugin-emoji/web/src/toolbar/insert-buttons.jsx [hunk 5]
```diff
@@ -11,36 +11,31 @@ import EmojiPluginIcon from '../icons/EmojiPluginIcon.svg';
...
       modalElement: decorateComponentWithProps(EmojiPreviewModal, {
         getEditorState,
         setEditorState,
         ...settings,
       }),
       modalStylesFn: ({ buttonRef }) => {
-        const modalStyles = getModalStyles({
+        const { modalStyles, contentStyles } = getBottomModalStyles(buttonRef, {
           customStyles: DesktopFlyOutModalStyles,
           fullScreen: true,
```
Comment (shaulgo) [inline packages/plugin-emoji/web/src/toolbar/insert-buttons.jsx]: if fullscreen is always true. It shouldn't be a pram

File: packages/plugin-emoji/web/src/toolbar/insert-buttons.jsx [hunk 6]
```diff
@@ -11,36 +11,31 @@ import EmojiPluginIcon from '../icons/EmojiPluginIcon.svg';
...
+
   return [
     {
       type: 'modal',
       name: 'EMOJI',
       tooltipText: t('EmojiPlugin_InsertButton_Tooltip'),
       Icon: icon,
       componentData: settings.componentDataDefaults || {},
-      toolbars: settings.insertToolbars || [TOOLBARS.FOOTER],
+      toolbars: [TOOLBARS.FOOTER],
```
Comment (shaulgo) [inline packages/plugin-emoji/web/src/toolbar/insert-buttons.jsx]: remind me what is settings.insertToolbars

File: packages/plugin-emoji/web/src/toolbar/insert-buttons.jsx [hunk 7]
```diff
@@ -2,7 +2,7 @@ import { DesktopFlyOutModalStyles } from '../constants';
 import {
   TOOLBARS,
   decorateComponentWithProps,
-  getModalStyles,
+  getBottomModalStyles,
```
Comment (shaulgo) [inline packages/plugin-emoji/web/src/toolbar/insert-buttons.jsx]: change here as well

File: packages/plugin-giphy/web/src/toolbar/insert-giphy-buttons.jsx [hunk 8]
```diff
@@ -27,30 +28,16 @@ export default ({ helpers, t, settings, isMobile }) => {
...
-        let modalLeft, modalRight;
-        if (isRtl) {
-          modalRight = window.innerWidth - right - 8;
-        } else {
-          modalLeft = left - 15;
-        }
-        const isAboveButton = top - 365 > 0;
-        const modalTop = isAboveButton ? top - 365 : top + 30;
         return {
           ...modalStyles,
```
Comment (shaulgo) [inline packages/plugin-giphy/web/src/toolbar/insert-giphy-buttons.jsx]: same as above

File: packages/plugin-emoji/web/src/constants.js (hunk 3 above)
Comment (barboaron) [reply packages/plugin-emoji/web/src/constants.js]: yes, it looks weird with the  'solid 1px #ededed' with the arrow icon style. I fix it according to the giphy plugin

File: packages/plugin-emoji/web/src/toolbar/insert-buttons.jsx (hunk 4 above)
Comment (barboaron) [reply packages/plugin-emoji/web/src/toolbar/insert-buttons.jsx]: yes, I will do this

**Pull request merged**